Makefile

```diff
@@ -51,7 +51,8 @@ tre-deploy:
 	&& . ./devops/scripts/check_dependencies.sh nodocker \
 	&& . ./devops/scripts/load_env.sh ./devops/terraform/.env \
 	&& . ./devops/scripts/load_env.sh ./templates/core/terraform/.env \
-	&& cd ./templates/core/terraform/ && ./deploy.sh
+	&& cd ./templates/core/terraform/ && ./deploy.sh \
+	&& cd ../../../ && ./devops/scripts/create_sp_for_deployer.sh
```

Comment:
Do we want a new SP creating each time we deploy? Do we need `make create_xxxx_sp` or something that is run separately?

Reply:
Also the Dev running make tre-deploy, might not have create SP permissions.

Reply:
ATM, no new SP is created each time, but if one exists, its secret will get updated. I could add a check in the script that if the SP exists, no action is taken.

Reply:
That would be good. 

However the scenario where dev can't create SPs - this is common on our projects (And even multiple deployments/TREs share the SP) - this needs to be catered for - so a separate step creates the SP. 

The deploy step then consumes the pre created credentials. This is the same SP that we have `ARM_xxx` variables for at present?

Reply:
How would you preserve the pre-created credentials? I had the SP creation in another step earlier, but it since there can be different processes running depending on the environment, the env variables don't always keep.

Reply:
This resolves as per separate discussion.